addons/bus/models/bus_listener_mixin.py

```diff
@@ -15,12 +15,15 @@ class BusListenerMixin(models.AbstractModel):
     def _bus_send(self, notification_type, message, /, *, subchannel=None):
         """Send a notification to the webclient."""
         for record in self:
-            main_channel = record._bus_channel()
+            main_channel = record
+            while (new_main_channel := main_channel._bus_channel()) != main_channel:
```

Comment:
Note: this allow `_bus_channel` to return "intermediate" records (we want to return user whenever possible rather than partner). The last step here will call `_bus_channel` on user to get the partner, and then on partner itself to stop the loop.